cmd/collector/app/zipkin/http_handler.go

```diff
@@ -92,7 +92,7 @@ func (aH *APIHandler) saveSpans(w http.ResponseWriter, r *http.Request) {
 	var tSpans []*zipkincore.Span
 	switch contentType {
 	case "application/x-thrift":
-		tSpans, err = zipkin.DeserializeThrift(bodyBytes)
+		tSpans, err = zipkin.DeserializeThrift(r.Context(), bodyBytes)
```

Comment:
what's the point of passing context here? DeserializeThrift is not an IO-bound function.

Reply:
First there was 
> Function `DeserializeThrift` should pass the context parameter (contextcheck)

If I just use
> zipkin.DeserializeThrift(context.Background(), bodyBytes)
I get
> Non-inherited new context, use function like `context.WithXXX` or `r.Context` instead (contextcheck)

Reply:
I see the problem: https://github.com/jaegertracing/jaeger/blob/74d6a1250ee5c9db6aa776510927e36fe5869681/model/converter/thrift/zipkin/deserialize.go#L40-L41

That's why it's flagging this specific function vs. other functions. It's a false positive because all serialization happens in-memory, without IO, but the way Thrift APIs are defined it looks like there is an IO and requires a context.